docs/prompting/thought_generation/chain_of_thought_zero_shot/step_back_prompting.md

```diff
@@ -1,7 +1,84 @@
 ---
-title: ""
-description: ""
-keywords: ""
+title: "Step-Back Prompting"
+description: "Step-back prompting is a two-step prompting technique that asks the LLM a step-back question to gather context for the query"
 ---
 
-[wip]
+# Step-Back Prompting
+
+Step-Back prompting<sup><a href="https://arxiv.org/abs/2310.06117">1</a></sup> is a variation of Chain of Thought prompting. The LLM is prompted in two steps:
+
+1. **Abstraction**: Ask the LLM a generic, higher-level concept. This is generally topic-specific. This is known as the _step-back question_.
+2. **Reasoning**: Ask the LLM the original question, given its answer to the abstract question. This is known as _abstracted-grounded reasoning_.
+
+!!! example "Step-Back Prompting Example"
+
+    **Original Question**: What happens to the pressure of an ideal gas when temperature and volume are increased?
+
+    **Step-Back Prompt**: What are the physics concepts associated with this question?
+
+    **Reasoning Prompt**: {step-back response} {original question}
+
+This has been shown to improve scores on reasoning benchmarks for PaLM-2L and GPT-4.<sup><a href="https://arxiv.org/abs/2406.06608">\*</a></sup>
+
+```python
+import openai
+import instructor
+from pydantic import BaseModel
+from typing import Iterable, Literal
+
+client = instructor.from_openai(openai.OpenAI())
+
+class Education(BaseModel):
+    degree: Literal["Bachelors", "Masters", "PhD"]
+    school: str
+    topic: str
+    year: int
+
+class Response(BaseModel):
+    school: str
+
+# Step 1: Ask the stepback question
+stepback = client.chat.completions.create(
```

Comment:
shouldn't we have a prompt that generates the abstract question? not pass in the abstract question?

Reply:
re the paper:

> The step-back question “What was Estella Leopold’s education history" in Figure 2 is generated by the LLM